runtime/src/main/cpp/FinalizerHooksTest.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#include "FinalizerHooks.hpp"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include "FinalizerHooksTestSupport.hpp"
+#include "Memory.h"
+
+using namespace kotlin;
+
+using ::testing::_;
+
+namespace {
+
+class FinalizerHooksTest : public testing::Test {
+public:
+    testing::MockFunction<void(ObjHeader*)>& finalizerHook() { return finalizerHooks_.finalizerHook(); }
+
+private:
+    FinalizerHooksTestSupport finalizerHooks_;
```

Comment:
(minor) Why not include the logic from `FinalizerHooksTestSupport` directly into `FinalizerHooksTest`? Will we need this logic in other tests?

Reply:
> Will we need this logic in other tests?

Yep, I think tests on GC might need that.